APMrover2/mode_acro.cpp

```diff
@@ -0,0 +1,40 @@
+#include "mode.h"
+#include "Rover.h"
+
+void ModeAcro::update()
+{
+    float desired_steering, desired_throttle;
+    get_pilot_desired_steering_and_throttle(desired_steering, desired_throttle);
+
+    // convert pilot throttle input to desired speed (up to twice the cruise speed)
+    float target_speed = desired_throttle * 0.01f * calc_speed_max(g.speed_cruise, g.throttle_cruise * 0.01f);
+
+    // get speed forward
+    float speed;
+    if (!attitude_control.get_forward_speed(speed)) {
+        // no valid speed so stop
```

Comment:
`stop_vehicle()` might be reasonable here.

Reply:
thanks for the review, we need the estimated speed to be able to stop the vehicle though so can't go higher.